src/arch/simddetect.cpp

```diff
@@ -65,6 +75,8 @@ bool SIMDDetect::avx512BW_available_;
 bool SIMDDetect::fma_available_;
 // If true, then SSe4.1 has been detected.
 bool SIMDDetect::sse_available_;
+// If true, then Neon has been detected.
+bool SIMDDetect::neon_available_;
```

Comment:
```suggestion
#if defined(HAVE_NEON)
// If true, then Neon has been detected.
bool SIMDDetect::neon_available_;
#endif
```

Reply:
I don't think I should do that.

None of the other things (avx_available_ etc) are guarded in that way. Also, that would stop IsNEONAvailable() being available in all builds, which would appear to be against the way things are done?

Reply:
NEON is only available for ARM, never for Intel, right? So it does not seam useful to add a NEON related variable to Intel code, and it also does not make sense to add AVX or SSE related variables to ARM code.

The way things are done is not constant, but changes when necessary or reasonable.